torch/nn/modules/upsampling.py

```diff
@@ -109,6 +111,104 @@ class UpsamplingBilinear2d(_UpsamplingBase):
         [torch.FloatTensor of size 1x1x4x4]
 
     """
+    def __init__(self, size=None, scale_factor=None):
+        super(UpsamplingBilinear2d, self).__init__(_pair(size), scale_factor)
 
     def forward(self, input):
         return F.upsample_bilinear(input, self.size, self.scale_factor)
+
+
+class UpsamplingNearest3d(_UpsamplingBase):
+    """
+    Applies a 3D nearest neighbor upsampling to an input signal composed of several input
+    channels.
+
+    To specify the scale, it takes either the :attr:`size` or the :attr:`scale_factor`
+    as it's constructor argument.
+
+    When `size` is given, it is the output size of the volume (d, h, w).
+
+    Args:
+        size (tuple, optional): a tuple of ints (D_out, H_out, W_out) output sizes
+        scale_factor (int, optional): the multiplier for the image depth / height / width
+
+    Shape:
+        - Input: :math:`(N, C, D_{in}, H_{in}, W_{in})`
+        - Output: :math:`(N, C, D_{out}, H_{out}, W_{out})` where
+          :math:`D_{out} = floor(D_{in} * scale\_factor)`
+          :math:`H_{out} = floor(H_{in} * scale\_factor)`
+          :math:`W_{out} = floor(W_{in}  * scale\_factor)`
+
+    Examples::
+
+        >>> inp
+        Variable containing:
+        (0 ,0 ,.,.) =
+          1  2
+          3  4
+        [torch.FloatTensor of size 1x1x2x2x2]
+
+        >>> m = nn.UpsamplingNearest3d(scale_factor=2)
+        >>> m(inp)
+        Variable containing:
+        (0 ,0 ,.,.) =
+          1  1  2  2
+          1  1  2  2
+          3  3  4  4
+          3  3  4  4
+        [torch.FloatTensor of size 1x1x4x4x4]
+
+    """
+    def __init__(self, size=None, scale_factor=None):
+        super(UpsamplingNearest3d, self).__init__(_triple(size), scale_factor)
```

Comment:
pass in size instead of _triple(size), and cast self.size() after the parent constructor